src/Type/MixedType.php

```diff
@@ -470,6 +470,17 @@ public function isArray(): TrinaryLogic
 		return TrinaryLogic::createMaybe();
 	}
 
+	public function isConstantArray(): TrinaryLogic
+	{
+		if ($this->subtractedType !== null) {
+			if ($this->subtractedType->isSuperTypeOf(new ConstantArrayType([], []))->yes()) {
+				return TrinaryLogic::createNo();
+			}
+		}
```

Comment:
supported subtraction is a new feature which should be covered by a test

Reply:
thx, I had a bad feeling about this and should have be more careful. trying to add tests made me realize that I have no clue how to implement this correctly for constant arrays, so I got rid of it and it returns always maybe now

Reply:
hmm just guesswork, but maybe something like

```
function doFoo($mixed) {
  $arr = ['a' => 1];
  if ($mixed != $arr) {
     assertType(...);
  }
}
```

Reply:
more the latter. I think my initial implementation with the empty constant array is not correct